internal/store/postgres/metaschema_repository.go

```diff
@@ -0,0 +1,303 @@
+		}
+
+		return m.dbc.SelectContext(ctx, &schemaModels, query, params...)
+	}); err != nil {
+		err = checkPostgresError(err)
+		if errors.Is(err, sql.ErrNoRows) {
+			return []metaschema.MetaSchema{}, metaschema.ErrNotExist
+		}
+		return []metaschema.MetaSchema{}, fmt.Errorf("%w: %s", dbErr, err)
+	}
+
+	var transformedSchemas []metaschema.MetaSchema
+	for _, m := range schemaModels {
+		transformedSchemas = append(transformedSchemas, m.tranformtoMetadataSchema())
+	}
+
+	return transformedSchemas, nil
+}
+
+func (m MetaSchemaRepository) Delete(ctx context.Context, name string) error {
+	query, params, err := dialect.Delete(TABLE_METASCHEMA).
+		Where(
+			goqu.Ex{
+				"name": name,
+			},
+		).ToSQL()
+	if err != nil {
+		return fmt.Errorf("%w: %s", queryErr, err)
+	}
+
+	return m.dbc.WithTimeout(ctx, func(ctx context.Context) error {
+		nrCtx := newrelic.FromContext(ctx)
+		if nrCtx != nil {
+			nr := newrelic.DatastoreSegment{
+				Product:    newrelic.DatastorePostgres,
+				Collection: TABLE_METASCHEMA,
+				Operation:  "Delete",
+				StartTime:  nrCtx.StartSegmentNow(),
+			}
+			defer nr.End()
+		}
+
+		result, err := m.dbc.ExecContext(ctx, query, params...)
+		if err != nil {
+			err = checkPostgresError(err)
+			switch {
+			case errors.Is(err, sql.ErrNoRows):
+				return fmt.Errorf("%w: %s", dbErr, metaschema.ErrNotExist)
+			default:
+				return fmt.Errorf("%w: %s", dbErr, err)
+			}
+		}
+
+		if count, _ := result.RowsAffected(); count > 0 {
+			delete(metaSchemaCache, name)
+			return nil
+		}
+
+		return metaschema.ErrNotExist
+	})
+}
+
+func (m MetaSchemaRepository) Update(ctx context.Context, name string, mschema metaschema.MetaSchema) (string, error) {
+	query, params, err := dialect.Update(TABLE_METASCHEMA).Set(
+		goqu.Record{
+			"schema": mschema.Schema,
+		}).Where(goqu.Ex{
+		"name": name,
+	}).Returning("name").ToSQL()
+
+	if err != nil {
+		return "", fmt.Errorf("%w: %s", queryErr, err)
+	}
+
+	var schemaName string
+	if err = m.dbc.WithTimeout(ctx, func(ctx context.Context) error {
+		nrCtx := newrelic.FromContext(ctx)
+		if nrCtx != nil {
+			nr := newrelic.DatastoreSegment{
+				Product:    newrelic.DatastorePostgres,
+				Collection: TABLE_METASCHEMA,
+				Operation:  "Update",
+				StartTime:  nrCtx.StartSegmentNow(),
+			}
+			defer nr.End()
+		}
+
+		return m.dbc.QueryRowxContext(ctx, query, params...).Scan(&schemaName)
+	}); err != nil {
+		err = checkPostgresError(err)
+		return "", fmt.Errorf("%w: %s", dbErr, err)
+	}
+
+	metaSchemaCache[mschema.Name] = mschema.Schema
+	return schemaName, nil
+}
+
+func (m MetaSchemaRepository) InitMetaSchemas(ctx context.Context) error {
+	// load schemas in memory from the db
+	schemas, err := m.List(ctx)
+	if err != nil {
+		return errors.New("error in initialising metadata json-schemas")
+	}
+
+	for _, s := range schemas {
+		metaSchemaCache[s.Name] = s.Schema
+	}
+
+	// if default metaschemas doesnt exists add to the db
+	if metaSchemaCache[userMetaSchemaName] == "" {
```

Comment:
instead of writing this 4 times, what if we create map with schema name and schema value and iterate over it?